ble/Gap.h

```diff
@@ -57,7 +57,7 @@ class Gap {
      * deprecated state to transparenly support existing applications which may
      * have used Gap::ADDR_TYPE_*.
      */
-    enum {
+    enum AddressTypeOverload_t {
```

Comment:
@LiyouZhou The whole idea of making this enum anonymous is to deprecate it. What exactly is the situation that you would like to support?